### PR TITLE
Support the `AUTO_RANDOM` Field

### PR DESCRIPTION
**Pull request by Icemap**
- [x] Do only one thing
- [x] Non breaking API changes
- [x] Tested

### What did this pull request do?

For this issue: https://github.com/go-gorm/gorm/issues/5977
Support the `AUTO_RANDOM`.

### User Case Description

If you already deploy a TiDB at `localhost:4000`. You can use this code to test.

```go
package main

import (
	"fmt"
	"gorm.io/driver/mysql"
	"gorm.io/gorm"
)

type Product struct {
	ID    uint `gorm:"primaryKey;default:auto_random()"`
	Code  string
	Price uint
}

func main() {
	db, err := gorm.Open(mysql.Open("root:@tcp(127.0.0.1:4000)/test"), &gorm.Config{})
	if err != nil {
		panic("failed to connect database")
	}

	db.AutoMigrate(&Product{})

	insertProduct := &Product{Code: "D42", Price: 100}

	db.Create(insertProduct)
	fmt.Printf("insert ID: %d, Code: %s, Prict: %d\n",
		insertProduct.ID, insertProduct.Code, insertProduct.Price)

	readProduct := &Product{}
	db.First(&readProduct, "code = ?", "D42") // find product with code D42

	fmt.Printf("read ID: %d, Code: %s, Prict: %d\n",
		readProduct.ID, readProduct.Code, readProduct.Price)
}

```

### CODE REVIEW

**Comment by Icemap** [comment]
Change these two PR's implementation methods:
https://github.com/go-gorm/mysql/pull/102
https://github.com/go-gorm/gorm/pull/5981
For avoiding interface changes.

**Comment by Icemap** [comment]
PTAL @jinzhu

File: mysql.go [hunk 1]
```diff
@@ -417,7 +421,32 @@ func (dialector Dialector) getSchemaBytesType(field *schema.Field) string {
...
+// 1. Is PrimaryKey;
+// 2. Has default value;
+// 3. Default value is "auto_random()";
+// 4. IGNORE the field.Size, it MUST be bigint;
+// 5. CLEAR the default tag, and return true;
+// 6. Otherwise, return false.
+func autoRandomType(field *schema.Field) (bool, string) {
+	if field.PrimaryKey && field.HasDefaultValue &&
+		strings.ToLower(strings.TrimSpace(field.DefaultValue)) == AutoRandomTag {
+		field.HasDefaultValue = false
```
Comment (jinzhu) [inline mysql.go:436]: Why do we need to set `HasDefaultValue` to false, it should be true?

Comment (Icemap) [reply mysql.go:436]: We used this tag `default:auto_random()` to implement the `auto_random` feature. But it is actually not a default value. It's a field like `auto_increment`.

Comment (jinzhu) [reply mysql.go:436]: for `auto_increment` fields we also marked them as `HasDefaultValue`, e.g:

https://github.com/go-gorm/gorm/blob/d834dd60b715422dc2a900fb2744f9c278a9830f/schema/field.go#L113

Comment (Icemap) [reply mysql.go:436]: Ah, got it. I made a mistake.

Comment (Icemap) [reply mysql.go]: Fixed. PTAL again.

**Pull request merged**